### ISSUE TITLE
Problem using custom projection EPSG:28992

### ISSUE THREAD

**Issue by justb4** [openstreetmap/osm2pgsql#1672]
## What version of osm2pgsql are you using?
```
osm2pgsql version 1.6.0
Build: None
Compiled using the following library versions:
Libosmium 2.17.3
Proj [API 6] 8.2.1
Lua 5.3.6
```

## What operating system and PostgreSQL/PostGIS version are you using?

Ubuntu Jammy (22.04) Docker Image.

## Tell us something about your system

16GB 8core Docker Desktop on MAcOSX.

## What did you do exactly?

osm2pgsql with custom projection EPSG:28992 via `--proj=28992`.

## What did you expect to happen?

Data in PostGIS in EPSG:28992. 

## What did happen instead?

Data in PostGIS ok with EPSG:28992 but coordinates are completely wrong, look like still in EPSG:3857, Web Merc.

## What did you do to try analyzing the problem?

Spent quite some time. Bashed into Docker Container, installed `proj-bin` since I suspect something wrong there.
```
$ projinfo EPSG:28992
PROJ.4 string:
+proj=sterea +lat_0=52.1561605555556 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +units=m +no_defs +type=crs

```
This seems wrong, it is missing the `towgs84` parm. Hard to analyze as there is no `epsg` file anymore.
The right proj string as I use in e.g. Mapnik,  should be:
`<!ENTITY srs28992 "+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +towgs84=565.417,50.3319,465.552,-0.398957,0.343988,-1.8774,4.0725 +units=m +no_defs">
`
Maybe a proj issue after all, or is there something I can patch?

**Comment by joto** [comment]
Please give us at least the exact command line you were using and any config files used. That's why there is the question "What did you do exactly?".

**Comment by justb4** [comment]
Ok, thanks @joto. Sorry, more info here, but as all was working fine in EPSG:3857 including Mapnik rendering, but went wrong with EPSG:28992 (invalid coordinate values in PostGIS). Steps taken:

- downloaded Netherlands North-Holland pbf from geofabrik.de, cut out small area with Osmosis as ${DOWNLOAD_FILE}
- run:
`osm2pgsql -U ${POSTGRES_USER} -d ${POSTGRES_DB} -H ${POSTGRES_HOST} -P ${POSTGRES_PORT} -s -S ${OSM2PGSQL_STYLE} ${DOWNLOAD_FILE}`
 - OSM2PGSQL_STYLE: Style file is standard OpenTopoMap Style
 - running the OpenTopoMap SQL files after
 - this worked fine, could generate Mapnik image with the OpenTopoMap Mapnik Style, no problems at all.
 - then trying to use custom projection EPSG:28992 (Dutch)
 - `osm2pgsql -E 28992 -U ${POSTGRES_USER} -d ${POSTGRES_DB} -H ${POSTGRES_HOST} -P ${POSTGRES_PORT} -s -S ${OSM2PGSQL_STYLE} ${DOWNLOAD_FILE}`
 - (or with `--proj=28992` same result).
 - database tables (PG 14 with PostGIS 3.2 in postgis:postgis Docker, also proj 8.2) filled, geometry columns have SRID 28992 ok, but all coordinates are not even in EPSG:28992 range (but look like EPSG:3857 range in NL but negative values)
 - like said I think the problem may be in proj, as projinfo EPSG:28992 does not contain the `+towgs84=565.2369,50.0087,465.658,-0.406857,0.350733,-1.87035,4.0812` part normally found, line in proj v5 in the `epsg` file. Full proj-string (from v5.2.0) is:
 `+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +towgs84=565.2369,50.0087,465.658,-0.406857,0.350733,-1.87035,4.0812 +units=m +no_defs`

Unless the `+towgs84` which is essential for any transform from/to WGS84 and other projections, somewhere else in proj. This happened more often in older proj/GDAL/PostGIS versions but we could always fix in the `epsg` file.
I will ask on the Proj Gitter chat first. I may be missing something essential.

**Comment by joto** [comment]
There was a bug in osm2pgsql when using modern Proj libraries. This was fixed in 4adcbde1 but it is not in a released version yet. Your proj version (8.2) is newer than this, so it might be that this is the cause of your problem. I just tested with master and I don't see a problem. Please try with master and see whether this fixes things for you.

**Comment by justb4** [comment]
Many thanks @joto ! I can confirm that reprojection is working for e.g. EPSG:28992 with a custom build of `osm2pgsql master` both with Proj 6 and 8. Builds/runs done within a Docker Image.

With Ubuntu 20.04 Docker Image:
```
osm2pgsql version 1.6.0
Build: RelWithDebInfo
Compiled using the following library versions:
Libosmium 2.17.3
Proj [API 4] Rel. 6.3.1, February 10th, 2020
Lua 5.3.3
```
With Ubuntu 22.04 Docker Image:

```
osm2pgsql version 1.6.0
Build: RelWithDebInfo
Compiled using the following library versions:
Libosmium 2.17.3
Proj [API 6] 8.2.1
Lua 5.3.6
```

So closing this issue, thanks again.
